packages/vector_graphics_compiler/lib/src/svg/parser.dart

```diff
@@ -2338,4 +2335,55 @@ class SvgFillAttributes {
 
     return Fill(color: resolvedColor, shader: shader);
   }
+
+  @override
+  String toString() {
+    return 'SvgFillAttributes('
+        'definitions: $_definitions, '
+        'color: $color, '
+        'shaderId: $shaderId, '
+        'hasPattern: $hasPattern, '
+        'oapctiy: $opacity)';
+  }
+}
+
+/// Represents a color for filling or stroking.
+///
+/// If the [color] is not null, [isNone] is false.
+///
+/// If the [color] is null and [isNone] is false, color should be inherited from
+/// the parent or defaulted as per the SVG specification.
+///
+/// If the [color] is null and [isNone] is true, inheritence should be cleared
+/// and no painting should happen.
+class ColorOrNone {
```

Comment:
This seems like a great way to abstract it but damn.

![image](https://user-images.githubusercontent.com/8975114/223298895-e26734e3-b9c6-4f6e-879d-e9a828d24e2f.png)